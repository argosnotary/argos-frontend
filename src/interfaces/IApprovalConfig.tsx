/*
 * Copyright (C) 2020 Argos Notary
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export interface IApprovalConfig {
  stepName: string;
  segmentName: string;
  artifactCollectorSpecifications: Array<IArtifactCollector>;
}

export enum ArtifactCollectorType {
  XLDEPLOY = "XLDEPLOY",
  GIT = "GIT"
}

export interface IArtifactCollector {
  name: string;
  type: ArtifactCollectorType;
  uri: string;
  context: IXLDeployContext | IGitContext;
}

export interface IXLDeployContext {
  applicationName: string;
}

export interface IGitContext {
  repository: string;
}
