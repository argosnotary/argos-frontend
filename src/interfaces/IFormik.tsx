/*
 * Copyright (C) 2020 Argos Notary
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
  FormikErrors,
  FormikTouched,
  FormikValues,
  FieldInputProps,
  FieldMetaProps,
  FieldHelperProps,
  FormikState
} from "formik";

interface IFormik<Values extends FormikValues> {
  initialValues: Values;
  initialErrors: FormikErrors<unknown>;
  initialTouched: FormikTouched<unknown>;
  initialStatus: any;
  handleBlur: (eventOrString: any) => void | ((e: any) => void);
  handleChange: (
    eventOrPath: string | React.ChangeEvent<any>
  ) => void | ((eventOrTextValue: string | React.ChangeEvent<any>) => void);
  handleReset: (e: any) => void;
  handleSubmit: (e?: React.FormEvent<HTMLFormElement> | undefined) => void;
  resetForm: (nextState?: Partial<FormikState<Values>> | undefined) => void;
  setErrors: (errors: FormikErrors<Values>) => void;
  setFormikState: (
    stateOrCb:
      | FormikState<Values>
      | ((state: FormikState<Values>) => FormikState<Values>)
  ) => void;
  setFieldTouched: (
    field: string,
    touched?: boolean,
    shouldValidate?: boolean | undefined
  ) => any;
  setFieldValue: (
    field: string,
    value: any,
    shouldValidate?: boolean | undefined
  ) => any;
  setFieldError: (field: string, value: string | undefined) => void;
  setStatus: (status: any) => void;
  setSubmitting: (isSubmitting: boolean) => void;
  setTouched: (
    touched: FormikTouched<Values>,
    shouldValidate?: boolean | undefined
  ) => any;
  setValues: (values: Values, shouldValidate?: boolean | undefined) => any;
  submitForm: () => Promise<void | undefined>;
  validateForm: (values?: Values) => Promise<FormikErrors<Values>>;
  validateField: (name: string) => Promise<void> | Promise<string | undefined>;
  isValid: boolean;
  dirty: boolean;
  unregisterField: (name: string) => void;
  registerField: (name: string, { validate }: any) => void;
  getFieldProps: (nameOrOptions: any) => FieldInputProps<any>;
  getFieldMeta: (name: string) => FieldMetaProps<any>;
  getFieldHelpers: (name: string) => FieldHelperProps<any>;
  validateOnBlur: boolean;
  validateOnChange: boolean;
  validateOnMount: boolean;
  values: Values;
  errors: FormikErrors<Values>;
  touched: FormikTouched<Values>;
  isSubmitting: boolean;
  isValidating: boolean;
  status?: any;
  submitCount: number;
}

export default IFormik;
