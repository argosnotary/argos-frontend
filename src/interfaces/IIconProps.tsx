export default interface IIconProps {
    color: string;
    size: number;
    transform?: string;
    onClick?: () => void;
}