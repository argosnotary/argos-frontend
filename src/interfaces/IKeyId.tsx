export interface IKeyId {
  keyId: string;
}
