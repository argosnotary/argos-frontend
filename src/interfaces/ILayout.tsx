/*
 * Copyright (C) 2019 - 2020 Rabobank Nederland
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export interface ILayoutMetaBlock {
  signatures: ISignature[];
  layout: ILayout;
}

export interface ISignature {
  keyId: string;
  signature: string;
  keyAlgorithm: string;
  hashAlgorithm: string;
}

export interface ILayout {
  keys: IPublicKey[];
  authorizedKeyIds: string[];
  expectedEndProducts: IRule[];
  layoutSegments: ILayoutSegment[];
}

export interface IPublicKey {
  keyId: string;
  publicKey: string;
}

export interface ILayoutSegment {
  name: string;
  steps: Array<IStep>;
}

export interface IStep {
  name: string;
  authorizedKeyIds: string[];
  requiredNumberOfLinks: number;
  expectedMaterials: IRule[];
  expectedProducts: IRule[];
}

export interface IRule {
  ruleType?: RuleRuleTypeEnum;
  pattern: string;
  sourcePathPrefix?: string;
  destinationPathPrefix?: string;
  destinationSegmentName?: string;
  destinationType?: RuleDestinationTypeEnum;
  destinationStepName?: string;
}

export enum RuleRuleTypeEnum {
  ALLOW = "ALLOW",
  CREATE = "CREATE",
  CREATE_OR_MODIFY = "CREATE_OR_MODIFY",
  DELETE = "DELETE",
  DISALLOW = "DISALLOW",
  MATCH = "MATCH",
  MODIFY = "MODIFY",
  REQUIRE = "REQUIRE"
}

export enum RuleDestinationTypeEnum {
  PRODUCTS = "PRODUCTS",
  MATERIALS = "MATERIALS"
}

export enum ILayoutValidationMessageTypes {
  DATA_INPUT = "DATA_INPUT",
  MODEL_CONSISTENCY = "MODEL_CONSISTENCY"
}

export interface ILayoutValidationMessage {
  field?: string;
  type: ILayoutValidationMessageTypes;
  message: string;
}
