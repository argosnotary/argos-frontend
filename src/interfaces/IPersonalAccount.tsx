import IRole from "./IRole";

interface IPersonalAccount {
  id: string;
  name: string;
  email: string;
  roles: Array<IRole>;
}

export default IPersonalAccount;
