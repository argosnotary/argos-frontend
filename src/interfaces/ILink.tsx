import {ISignature} from "./ILayout";

export interface ILink {
    runId: string;
    layoutSegmentName: string;
    stepName: string;
    command: string[];
    materials: IArtifact[];
    products: IArtifact[];
}

export interface IArtifact {
    uri: string;
    hash: string;
}

export interface ILinkMetaBlock {
    supplyChainId: string;
    signature: ISignature;
    link: ILink;
}