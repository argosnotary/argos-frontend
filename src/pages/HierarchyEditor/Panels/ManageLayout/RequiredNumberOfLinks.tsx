/*
 * Copyright (C) 2019 - 2020 Rabobank Nederland
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { useEffect } from "react";
import { FormPermissions } from "../../../../types/FormPermission";
import { IGenericFormSchema } from "../../../../interfaces/IGenericFormSchema";
import {
  LayoutEditorActionType,
  useLayoutEditorStore
} from "../../../../stores/LayoutEditorStore";
import useFormBuilder, {
  FormSubmitButtonHandlerTypes,
  IFormBuilderConfig
} from "../../../../hooks/useFormBuilder";
import { IStep } from "../../../../interfaces/ILayout";

const formSchema: IGenericFormSchema = [
  {
    labelValue: "Required Number Of Links*",
    name: "requiredNumberOfLinks",
    formType: "text"
  }
];

interface IFormValues {
  requiredNumberOfLinks: string;
}

const validateForm = (values: IFormValues) => {
  const errors = {} as IFormValues;

  if (!values.requiredNumberOfLinks) {
    errors.requiredNumberOfLinks = "Please fill in required number of links.";
  } else if (!new RegExp("^\\d+$").test(values.requiredNumberOfLinks)) {
    errors.requiredNumberOfLinks = "Please enter a number.";
  }
  return errors;
};

const RequiredNumberOfLinks: React.FC = () => {
  const editorStoreContext = useLayoutEditorStore();

  const updateRequiredNumberOfLinks = (values: IFormValues) => {
    editorStoreContext.dispatch({
      type: LayoutEditorActionType.UPDATE_REQUIRED_NUMBER_OF_LINKS,
      layoutStep: {
        requiredNumberOfLinks: parseInt(values.requiredNumberOfLinks)
      } as IStep
    });
  };

  const formConfig: IFormBuilderConfig = {
    dataTesthookId: "required-number-of-links-form",
    schema: formSchema,
    permission: FormPermissions.EDIT,
    isLoading: false,
    validate: validateForm,
    onSubmit: updateRequiredNumberOfLinks,
    onChange: (valid, values) => {
      if (valid) {
        updateRequiredNumberOfLinks(values);
      }
    },
    buttonHandler: FormSubmitButtonHandlerTypes.CLICK
  };

  const [formJSX, formApi] = useFormBuilder(formConfig);

  useEffect(() => {
    if (
      editorStoreContext.state.selectedLayoutElement &&
      editorStoreContext.state.selectedLayoutElement.step &&
      editorStoreContext.state.selectedLayoutElement.step.requiredNumberOfLinks
    ) {
      formApi.setInitialFormValues({
        requiredNumberOfLinks:
          "" +
          editorStoreContext.state.selectedLayoutElement.step
            .requiredNumberOfLinks
      });
    } else {
      formApi.setInitialFormValues({
        requiredNumberOfLinks: ""
      });
    }
  }, [editorStoreContext.state.selectedLayoutElement?.step?.requiredNumberOfLinks]);

  return <>{formJSX}</>;
};

export default RequiredNumberOfLinks;
