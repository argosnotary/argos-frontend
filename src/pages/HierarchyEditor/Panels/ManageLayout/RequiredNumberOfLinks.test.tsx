/*
 * Copyright (C) 2020 Argos Notary CoÃ¶peratie UA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from "react";
import { mount, ReactWrapper } from "enzyme";

import { act } from "react-dom/test-utils";
import { waitFor } from "@testing-library/dom";
import {
  DetailsPanelType,
  ILayoutEditorStoreContext,
  LayoutEditorStoreContext
} from "../../../../stores/LayoutEditorStore";
import { ILayout, IStep } from "../../../../interfaces/ILayout";
import theme from "../../../../theme/base.json";
import { ThemeProvider } from "styled-components";
import RequiredNumberOfLinks from "./RequiredNumberOfLinks";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn()
  })
}));

const dispatch = jest.fn();

const editorStoreContext: ILayoutEditorStoreContext = {
  state: {
    layout: {} as ILayout,
    detailPanelMode: DetailsPanelType.LAYOUT_DETAILS,
    approvalConfigs: [],
    showSigningDialog: false,
    loading: false,
    activeEditLayoutElement: undefined,
    selectedLayoutElement: { step: { requiredNumberOfLinks: 12 } as IStep },
    showJson: false
  },
  dispatch: dispatch
};

function createComponent() {
  return mount(
    <ThemeProvider theme={theme}>
      <LayoutEditorStoreContext.Provider value={editorStoreContext}>
        <RequiredNumberOfLinks />
      </LayoutEditorStoreContext.Provider>
    </ThemeProvider>
  );
}

it("renders correctly", async () => {
  const root = createComponent();

  await act(async () => {
    await waitFor(() => {
      root.update();
      return expect(
        root.find(
          'input[data-testhook-id="required-number-of-links-form-field-0"]'
        ).length
      ).toBe(1);
    });

    expect(root.find(RequiredNumberOfLinks)).toMatchSnapshot();
  });
});

it("update required number of links", async () => {
  const root = createComponent();

  await act(async () => {
    await waitFor(() => {
      root.update();
      return expect(
        root.find(
          'input[data-testhook-id="required-number-of-links-form-field-0"]'
        ).length
      ).toBe(1);
    });

    root
      .find('form[data-testhook-id="required-number-of-links-form"]')
      .simulate("submit");

    expect(root.find(RequiredNumberOfLinks)).toMatchSnapshot();

    updateField(
      root.find(
        'input[data-testhook-id="required-number-of-links-form-field-0"]'
      ),
      "requiredNumberOfLinks",
      "4"
    );

    root
      .find('form[data-testhook-id="required-number-of-links-form"]')
      .simulate("submit");

    await waitFor(() => {
      root.update();
      return expect(dispatch.mock.calls.length).toBe(2);
    });

    expect(dispatch.mock.calls[0][0]).toEqual({
      layoutStep: {
        requiredNumberOfLinks: 4
      },
      type: 30
    });
  });
});

const updateField = (wrapper: ReactWrapper<any>, name: string, value: any) => {
  wrapper.simulate("input", {
    persist: () => {},
    target: {
      name,
      value
    }
  });
  wrapper.simulate("change", {
    persist: () => {},
    target: {
      name,
      value
    }
  });
};
