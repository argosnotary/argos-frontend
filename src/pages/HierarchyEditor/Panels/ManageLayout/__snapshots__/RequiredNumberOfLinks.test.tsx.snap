// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RequiredNumberOfLinks>
  <Styled(styled.div)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-jzJRlG",
            "isStatic": false,
            "lastClassName": "jNQHJE",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
",
              "
  display: flex;
  flex-direction: column;
  width: 100%;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-gqjmRU",
          ],
          "render": [Function],
          "styledComponentId": "sc-jzJRlG",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-gqjmRU sc-jzJRlG jNQHJE"
      >
        <form
          data-testhook-id="required-number-of-links-form"
          onBlur={[Function]}
          onSubmit={[Function]}
        >
          <FormInput
            dataTesthookId="required-number-of-links-form-field-0"
            disabled={false}
            formType="text"
            innerRef={
              Object {
                "current": <input
                  class="sc-bdVaJa iSgtnE"
                  data-testhook-id="required-number-of-links-form-field-0"
                  name="requiredNumberOfLinks"
                  type="text"
                  value="12"
                />,
              }
            }
            labelValue="Required Number Of Links*"
            name="requiredNumberOfLinks"
            onBlur={[Function]}
            onChange={[Function]}
            value="12"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-htpNat",
                      "isStatic": false,
                      "lastClassName": "dePyzG",
                      "rules": Array [
                        "
  display: flex;
  flex-direction: column;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-htpNat",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-htpNat dePyzG"
                >
                  <styled.label>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bwzfXH",
                            "isStatic": false,
                            "lastClassName": "iGGvEu",
                            "rules": Array [
                              "
  font-size: 0.875rem;
  font-weight: normal;
  line-height: 1.8;
  color: #0a0a0a;
",
                            ],
                          },
                          "displayName": "styled.label",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bwzfXH",
                          "target": "label",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <label
                        className="sc-bwzfXH iGGvEu"
                      >
                        Required Number Of Links*
                      </label>
                    </StyledComponent>
                  </styled.label>
                  <styled.input
                    data-testhook-id="required-number-of-links-form-field-0"
                    disabled={false}
                    name="requiredNumberOfLinks"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value="12"
                  >
                    <StyledComponent
                      data-testhook-id="required-number-of-links-form-field-0"
                      disabled={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "iSgtnE",
                            "rules": Array [
                              "
  ",
                              "
  overflow: visible;
  box-sizing: border-box;
  height: ",
                              [Function],
                              ";
  margin: 0 0 1rem;
  padding: 0.5rem;
  border: 1px solid #cacaca;
  border-radius: 0;
  background-color: ",
                              [Function],
                              ";
  box-shadow: inset 0 1px 2px rgba(10, 10, 10, 0.1);
  font-family: inherit;
  font-size: 1rem;
  font-weight: normal;
  line-height: 1.5;
  color: #0a0a0a;
  transition: box-shadow 0.5s, border-color 0.25s ease-in-out,
    -webkit-box-shadow 0.5s;

  &:hover {
    cursor: ",
                              [Function],
                              ";
  }

  &:focus {
    outline: none;
    border: 1px solid #8a8a8a;
    box-shadow: 0 0 5px #cacaca;
  }
",
                              "
",
                            ],
                          },
                          "displayName": "styled.input",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "input",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={
                        Object {
                          "current": <input
                            class="sc-bdVaJa iSgtnE"
                            data-testhook-id="required-number-of-links-form-field-0"
                            name="requiredNumberOfLinks"
                            type="text"
                            value="12"
                          />,
                        }
                      }
                      name="requiredNumberOfLinks"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="text"
                      value="12"
                    >
                      <input
                        className="sc-bdVaJa iSgtnE"
                        data-testhook-id="required-number-of-links-form-field-0"
                        disabled={false}
                        name="requiredNumberOfLinks"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="text"
                        value="12"
                      />
                    </StyledComponent>
                  </styled.input>
                </div>
              </StyledComponent>
            </styled.div>
          </FormInput>
        </form>
      </div>
    </StyledComponent>
  </Styled(styled.div)>
</RequiredNumberOfLinks>
`;

exports[`update required number of links 1`] = `
<RequiredNumberOfLinks>
  <Styled(styled.div)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-jzJRlG",
            "isStatic": false,
            "lastClassName": "jNQHJE",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
",
              "
  display: flex;
  flex-direction: column;
  width: 100%;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-gqjmRU",
          ],
          "render": [Function],
          "styledComponentId": "sc-jzJRlG",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-gqjmRU sc-jzJRlG jNQHJE"
      >
        <form
          data-testhook-id="required-number-of-links-form"
          onBlur={[Function]}
          onSubmit={[Function]}
        >
          <FormInput
            dataTesthookId="required-number-of-links-form-field-0"
            disabled={false}
            formType="text"
            innerRef={
              Object {
                "current": <input
                  class="sc-bdVaJa iSgtnE"
                  data-testhook-id="required-number-of-links-form-field-0"
                  name="requiredNumberOfLinks"
                  type="text"
                  value="12"
                />,
              }
            }
            labelValue="Required Number Of Links*"
            name="requiredNumberOfLinks"
            onBlur={[Function]}
            onChange={[Function]}
            value="12"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-htpNat",
                      "isStatic": false,
                      "lastClassName": "dePyzG",
                      "rules": Array [
                        "
  display: flex;
  flex-direction: column;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-htpNat",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-htpNat dePyzG"
                >
                  <styled.label>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bwzfXH",
                            "isStatic": false,
                            "lastClassName": "iGGvEu",
                            "rules": Array [
                              "
  font-size: 0.875rem;
  font-weight: normal;
  line-height: 1.8;
  color: #0a0a0a;
",
                            ],
                          },
                          "displayName": "styled.label",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bwzfXH",
                          "target": "label",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <label
                        className="sc-bwzfXH iGGvEu"
                      >
                        Required Number Of Links*
                      </label>
                    </StyledComponent>
                  </styled.label>
                  <styled.input
                    data-testhook-id="required-number-of-links-form-field-0"
                    disabled={false}
                    name="requiredNumberOfLinks"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value="12"
                  >
                    <StyledComponent
                      data-testhook-id="required-number-of-links-form-field-0"
                      disabled={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "iSgtnE",
                            "rules": Array [
                              "
  ",
                              "
  overflow: visible;
  box-sizing: border-box;
  height: ",
                              [Function],
                              ";
  margin: 0 0 1rem;
  padding: 0.5rem;
  border: 1px solid #cacaca;
  border-radius: 0;
  background-color: ",
                              [Function],
                              ";
  box-shadow: inset 0 1px 2px rgba(10, 10, 10, 0.1);
  font-family: inherit;
  font-size: 1rem;
  font-weight: normal;
  line-height: 1.5;
  color: #0a0a0a;
  transition: box-shadow 0.5s, border-color 0.25s ease-in-out,
    -webkit-box-shadow 0.5s;

  &:hover {
    cursor: ",
                              [Function],
                              ";
  }

  &:focus {
    outline: none;
    border: 1px solid #8a8a8a;
    box-shadow: 0 0 5px #cacaca;
  }
",
                              "
",
                            ],
                          },
                          "displayName": "styled.input",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "input",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={
                        Object {
                          "current": <input
                            class="sc-bdVaJa iSgtnE"
                            data-testhook-id="required-number-of-links-form-field-0"
                            name="requiredNumberOfLinks"
                            type="text"
                            value="12"
                          />,
                        }
                      }
                      name="requiredNumberOfLinks"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="text"
                      value="12"
                    >
                      <input
                        className="sc-bdVaJa iSgtnE"
                        data-testhook-id="required-number-of-links-form-field-0"
                        disabled={false}
                        name="requiredNumberOfLinks"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="text"
                        value="12"
                      />
                    </StyledComponent>
                  </styled.input>
                </div>
              </StyledComponent>
            </styled.div>
          </FormInput>
        </form>
      </div>
    </StyledComponent>
  </Styled(styled.div)>
</RequiredNumberOfLinks>
`;
